web/src/user_group_edit.ts

```diff
@@ -314,32 +314,59 @@ function update_group_membership_button(group_id: number): void {
     }
 }
 
+function get_group_row_html(group: UserGroup): string {
+    const associated_subgroups = user_groups.get_associated_subgroups(
+        group,
+        people.my_current_user_id(),
+    );
+
+    const item: UserGroup & {
+        is_member: boolean;
+        can_join: boolean;
+        can_leave: boolean;
+        is_direct_member: boolean;
+        associated_subgroup_names: string;
+    } = {
+        is_member: user_groups.is_user_in_group(group.id, people.my_current_user_id()),
+        can_join: settings_data.can_join_user_group(group.id),
+        can_leave: settings_data.can_leave_user_group(group.id),
+        is_direct_member: user_groups.is_direct_member_of(people.my_current_user_id(), group.id),
+        associated_subgroup_names: user_groups.format_group_list(associated_subgroups),
+        ...group,
+    };
+    const html = render_browse_user_groups_list_item(item);
+    return html;
+}
+
 function update_display_checkmark_on_group_edit(group: UserGroup): void {
-    if (is_group_already_present(group)) {
-        const $row = row_for_group_id(group.id);
+    const tab_key = get_active_data().$tabs.first().attr("data-tab-key");
+    if (tab_key === "your-groups") {
+        // There is no need to do if "Your groups" tab is opened,
+        // because the whole list is already redrawn.
+        return;
+    }
 
-        const item = group;
-        const is_member = user_groups.is_user_in_group(group.id, people.my_current_user_id());
-        const can_join = settings_data.can_join_user_group(item.id);
-        const can_leave = settings_data.can_leave_user_group(item.id);
-        const is_direct_member = user_groups.is_direct_member_of(
-            people.my_current_user_id(),
-            item.id,
-        );
-        const associated_subgroups = user_groups.get_associated_subgroups(
-            item,
-            people.my_current_user_id(),
-        );
-        const associated_subgroup_names = user_groups.format_group_list(associated_subgroups);
-        const item_render_data = {
-            ...item,
-            is_member,
-            can_join,
-            can_leave,
-            is_direct_member,
-            associated_subgroup_names,
-        };
-        const html = render_browse_user_groups_list_item(item_render_data);
+    const current_user_id = people.my_current_user_id();
+
+    const $row = row_for_group_id(group.id);
+    const html = get_group_row_html(group);
+    const $new_row = $(html);
+
+    // TODO: Remove this if/when we just handle "active" when rendering templates.
+    if ($row.hasClass("active")) {
+        $new_row.addClass("active");
+    }
+
+    $row.replaceWith($new_row);
```

Comment:
Should we have L351-L360 in it's own function since it's getting repeated for supergroups also.
```
    const $row = row_for_group_id(group.id);
    const html = get_group_row_html(group);
    const $new_row = $(html);

    // TODO: Remove this if/when we just handle "active" when rendering templates.
    if ($row.hasClass("active")) {
        $new_row.addClass("active");
    }

    $row.replaceWith($new_row);
 ```